src/operators.jl

```diff
@@ -743,32 +743,11 @@ julia> collect(edges(h))
  Edge 3 => 4
 ```
 """
-function merge_vertices(g::AbstractGraph, vs)
-    labels = collect(1:nv(g))
-    # Use lowest value as new vertex id.
-    svs = sort!(vs)
-    nvnew = nv(g) - length(unique(svs)) + 1
-    nvnew <= nv(g) || return g
-    (v0, vm) = (first(svs), last(svs))
-    v0 > 0 || throw(ArgumentError("invalid vertex ID: $v0 in list of vertices to be merged"))
-    vm <= nv(g) || throw(ArgumentError("vertex $vm not found in graph")) # TODO 0.7: change to DomainError?
-    labels[svs] .= v0
-    shifts = compute_shifts(nv(g), svs[2:end])
-    for v in vertices(g)
-        if labels[v] != v0
-            labels[v] -= shifts[v]
-        end
-    end
-
-    #if v in svs then labels[v] == v0 else labels[v] == v
-    newg = SimpleGraph(nvnew)
-    for e in edges(g)
-        u, w = src(e), dst(e)
-        if labels[u] != labels[w] #not a new self loop
-            add_edge!(newg, labels[u], labels[w])
-        end
-    end
-    return newg
+function merge_vertices(graph::AbstractGraph, vs)
+    g = copy(graph)
```

Comment:
I thought about doing this but it doubles the memory requirements and will not work for large graphs.